clippy_lints/src/write.rs

```diff
@@ -234,20 +237,30 @@ fn check_tts<'a>(cx: &EarlyContext<'a>, tts: &ThinTokenStream, is_write: bool) -
         false,
         false,
     );
+    let mut expr: Option<Expr> = None;
     if is_write {
         // skip the initial write target
-        parser.parse_expr().map_err(|mut err| err.cancel()).ok()?;
+        expr = match parser.parse_expr().map_err(|mut err| err.cancel()).ok() {
+            Some(p) => Some(p.and_then(|expr| expr)),
+            None => return (None, None),
+        };
         // might be `writeln!(foo)`
-        parser.expect(&token::Comma).map_err(|mut err| err.cancel()).ok()?;
+        if let None = parser.expect(&token::Comma).map_err(|mut err| err.cancel()).ok() {
```

Comment:
You can change the `ok()` to `is_err()`. `if parser.expect().map_err().is_err() { ... }`